pkg/services/live/features/dashboard.go

```diff
@@ -28,16 +45,97 @@ func (h *DashboardHandler) GetHandlerForPath(path string) (models.ChannelHandler
+		}
+
+		dash := query.Result
+		guardian := guardian.New(dash.Id, user.OrgId, user)
+		if canView, err := guardian.CanView(); err != nil || !canView {
+			return models.SubscribeReply{}, backend.SubscribeStreamStatusPermissionDenied, nil
+		}
+
+		return models.SubscribeReply{
+			Presence:  true,
+			JoinLeave: true,
+		}, backend.SubscribeStreamStatusOK, nil
+	}
+
+	// Unknown path
+	logger.Error("Unknown dashboard channel", "path", e.Path)
+	return models.SubscribeReply{}, backend.SubscribeStreamStatusNotFound, nil
 }
 
 // OnPublish is called when someone begins to edit a dashboard
-func (h *DashboardHandler) OnPublish(ctx context.Context, _ *models.SignedInUser, e models.PublishEvent) (models.PublishReply, backend.PublishStreamStatus, error) {
-	return models.PublishReply{}, backend.PublishStreamStatusOK, nil
+func (h *DashboardHandler) OnPublish(ctx context.Context, user *models.SignedInUser, e models.PublishEvent) (models.PublishReply, backend.PublishStreamStatus, error) {
+	parts := strings.Split(e.Path, "/")
+	if parts[0] == "gitops" {
+		// gitops gets all changes for everything, so lets make sure it is an admin user
+		if !user.HasRole(models.ROLE_ADMIN) {
+			return models.PublishReply{}, backend.PublishStreamStatusPermissionDenied, nil
+		}
+
+		// Eventually this could broadcast a message back to the dashboard saying a pull request exists
+		return models.PublishReply{}, backend.PublishStreamStatusNotFound, fmt.Errorf("not implemented yet")
+	}
+
+	// make sure can view this dashboard
+	if len(parts) == 2 && parts[0] == "uid" {
+		event := dashboardEvent{}
+		err := json.Unmarshal(e.Data, &event)
+		if err != nil || event.UID != parts[1] {
+			return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, fmt.Errorf("bad request")
+		}
+		if event.Action != EDITING_STARTED {
+			// just ignore the event
+			return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, fmt.Errorf("ignore???")
+		}
+		query := models.GetDashboardQuery{Uid: parts[1], OrgId: user.OrgId}
+		if err := bus.Dispatch(&query); err != nil {
+			logger.Error("Unknown dashboard", "query", query)
+			return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, nil
+		}
+
+		guardian := guardian.New(query.Result.Id, user.OrgId, user)
+		canEdit, err := guardian.CanEdit()
+		if err != nil {
+			return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, fmt.Errorf("internal error")
+		}
+
+		// Ignore edit events if the user can not edit
+		if !canEdit {
+			return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, nil // NOOP
+		}
+
+		// Tell everyone who is editing
+		event.User = user.ToUserDisplayDTO()
+
+		msg, err := json.Marshal(event)
+		if err != nil {
+			return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, fmt.Errorf("internal error")
+		}
+		return models.PublishReply{Data: msg}, backend.PublishStreamStatusOK, nil
+	}
+
+	return models.PublishReply{}, backend.SubscribeStreamStatusNotFound, nil
```

Comment:
As we discussed in the morning: really too much args to return just a `NotFound`. Hope we eventually will come to
```
return nil, live.NotFound
```

or similar